File: packages/i18nify-go/modules/currency/get_iso_numeric_code.go
package currency

import "fmt"

// GetISONumericCode returns the ISO 4217 three-digit numeric code string for
// the given alphabetic currency code (e.g. "USD" → "840", "INR" → "356").
//
// The returned value is a zero-padded string as defined by ISO 4217
// (e.g. "008" for ALL, not "8"). It returns an error when the code is not
// found in the dataset.
func GetISONumericCode(currencyCode string) (string, error) {
	info, exists := cachedCurrencyData.CurrencyInformation[currencyCode]
	if !exists {
		return "", fmt.Errorf("invalid currency code: %q", currencyCode)
	}
	return info.NumericCode, nil
}

File: packages/i18nify-js/src/modules/currency/getISONumericCode.ts
import { withErrorBoundary } from '../../common/errorBoundary';
import NUMERIC_CODES from './data/numericCodes.json';
import { CurrencyCodeType } from './types';

/**
 * Returns the ISO 4217 three-digit numeric code for the given alphabetic
 * currency code (e.g. "USD" → "840", "INR" → "356").
 *
 * @param {CurrencyCodeType} currencyCode - A valid ISO 4217 alpha-3 code.
 * @returns {string} Zero-padded three-digit numeric code string (e.g. "008").
 * @throws When the currency code is not in the dataset.
 */
const getISONumericCode = (currencyCode: CurrencyCodeType): string => {
  if (!(currencyCode in NUMERIC_CODES)) {
    throw new Error(
      `Invalid currency code: "${String(currencyCode)}". Please provide a valid ISO 4217 alphabetic currency code.`,
    );
  }
  const numericCode = NUMERIC_CODES[currencyCode as keyof typeof NUMERIC_CODES];
  if (!numericCode) {
    throw new Error(
      `No numeric code found for currency: "${String(currencyCode)}".`,
    );
  }
  return numericCode;
};

export default /*#__PURE__*/ withErrorBoundary<typeof getISONumericCode>(
  getISONumericCode,
);
